src/main.c

```diff
@@ -629,6 +629,17 @@ int main(int argc, char **argv)
             for (int i = -10; i < (int) rc2[k].nruns; i++) {
 #define arr_len (1)
                 if (rc2[k].kernel == MULTISCATTER) {
+                  if (rc2[k].pattern_scatter_len > rc2[k].local_work_size) {
+                      error("Pattern length cannot exceed local_work_size", ERROR);
+                  }
+                  if (rc2[k].pattern_scatter_len > 1024) {
```

Comment:
A more accurate check may be
```
if (rc2[k].local_work_size > 1024)
```
since the local work size is the limiting factor. Eventually, if support for pattern_len > local_work_size is implemented, then checking pattern length wouldn't necessarily matter so much.
As an aside, would this be better caught by checking for errors from a kernel launch? If the requested local_work_size is larger than what is supported by the GPU, then each kernel would fail to launch. Either way, it's certainly a step in a good direction, it's just a matter of how much you want to implement now.

Reply:
A CUDA API error check call could maybe be added right before the `hipMemcpyFromSymbol` that happens after each of the kernels runs, that might be able to provide some coverage on checking the kernels.

Reply:
Yeah that's a good idea. I'll add an issue for that